pkg/occlient/occlient_test.go

```diff
@@ -406,3 +408,883 @@ func Test_parseImageName(t *testing.T) {
 		})
 	}
 }
+
+func TestClient_UpdateDCAnnotations(t *testing.T) {
+	tests := []struct {
+		name        string
+		dcName      string
+		annotations map[string]string
+		existingDc  appsv1.DeploymentConfig
+		wantErr     bool
+	}{
+		{
+			name:   "dc name with annotations",
+			dcName: "nodejs",
+			annotations: map[string]string{
+				"app.kubernetes.io/url":                   "file:///temp/nodejs-ex",
+				"app.kubernetes.io/component-source-type": "local",
+			},
+			existingDc: appsv1.DeploymentConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "nodejs",
+					Annotations: map[string]string{"app.kubernetes.io/url": "https://github.com/sclorg/nodejs-ex",
+						"app.kubernetes.io/component-source-type": "git",
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:   "dc name with annotations",
+			dcName: "nodejs",
+			annotations: map[string]string{
+				"app.kubernetes.io/url":                   "file:///temp/nodejs-ex",
+				"app.kubernetes.io/component-source-type": "local",
+			},
+			existingDc: appsv1.DeploymentConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "wildfly",
+					Annotations: map[string]string{"app.kubernetes.io/url": "https://github.com/sclorg/nodejs-ex",
+						"app.kubernetes.io/component-source-type": "git",
+					},
+				},
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fkclient, fkclientset := FakeNew()
+			fkclientset.AppsClientset.PrependReactor("get", "deploymentconfigs", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+				if tt.dcName != tt.existingDc.Name {
```

Comment:
Why is this check here?

Reply:
To check if the get on buildConfig was called with the right buildConfigName or not